npe2/manifest/schema.py

```diff
@@ -376,6 +379,87 @@ def _from_package_or_name(
 
     ValidationError = ValidationError  # for convenience of access
 
+    @classmethod
+    def _from_npe1_plugin(cls, plugin_name: str):
+        from .._from_npe1 import npe1_plugin_manager  # type: ignore
+
+        plugin_manager = npe1_plugin_manager()
+        if plugin_name not in plugin_manager.plugins:
+            # TODO: it would be nice to add some logic to prevent confusion here.
+            # for example... if the plugin name doesn't equal the package name, we
+            # should still be able to find it if the user gives a package name
+            from importlib.metadata import PackageNotFoundError, distribution
+
+            try:
+                dist = distribution(plugin_name)  # returns a list.  multiple plugins?
+                plugin_name = dist.entry_points[0].name
+            except PackageNotFoundError:
+                print(
+                    f"Could not find plugin {plugin_name!r}\n"
+                    f"Found {set(plugin_manager.plugins)}"
+                )
+
+        module = plugin_manager.plugins[plugin_name]
+        standard_meta = plugin_manager.get_standard_metadata(plugin_name)
+        package = standard_meta["package"].replace("-", "_")
+
+        commands = []
+        readers = []
+        writers = []
+        write_layers = []
+        for caller in plugin_manager._plugin2hookcallers[module]:
+            for impl in caller.get_hookimpls():
+                if impl.plugin_name != plugin_name:
+                    continue
+                name = impl.specname.replace("napari_", "")
+                id = f"{package}.{name}"
+                py_name = f"{impl.function.__module__}:{impl.function.__qualname__}"
+                title = " ".join(name.split("_")).title()
+                commands.append({"id": id, "python_name": py_name, "title": title})
+
+                if "reader" in id:
+                    # add this to the readers list
+                    readers.append({"command": id, "accepts_directories": True})
+
+                if "write" in id and "get_writer" not in id:
+                    # add this to the writers list
+                    layer = id.split("write_")[1]
+                    writers.append(
+                        {"command": id, "layer_types": [layer], "name": layer}
+                    )
+                    write_layers.append(layer + "*")
+
+                themes = []
+                if "theme" in id:
+                    for name, theme in impl.function():
+                        # cast theme into manifest.themes.ThemeColors dict
+                        manifest_theme = ThemeColors().dict()
+                        manifest_theme = {
+                            key: theme.get(key, manifest_theme[key])
+                            for key in manifest_theme
+                        }
```

Comment:
actually, I think you can just do `ThemeColors(**theme)` here

Reply:
ok, will change this